homeassistant/components/wemo/entity.py

```diff
@@ -58,16 +58,28 @@ async def async_update(self) -> None:
             return
 
         try:
-            with async_timeout.timeout(self.platform.scan_interval.seconds - 0.1):
-                await asyncio.shield(self._async_locked_update(True))
+            with async_timeout.timeout(
```

Comment:
Normally the timeout is used with an async context manager.
```py
async with async_timeout.timeout(...):
```
Is there a reason why we aren't doing that here?

Reply:
Nice catch. Thank you.

Fixed.